org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FoldingRangeHandler.java

```diff
@@ -116,11 +118,19 @@ private void computeFoldingRanges(List<FoldingRange> foldingRanges, ITypeRoot un
 							}
 						}
 						break;
+					case ITerminalSymbols.TokenNameimport:
+						startImport = startImport == -1 ? start : startImport;
+						endImport = scanner.getCurrentTokenEndPosition();

```

Comment:
Not mandatory, but would be nice if possible :

Look at `computeTypeRootRanges(..)` . We compute the import range for `.java` files there. It would be nice to compute the `.class` file import range there as well. The `scanner` has a method `resetTo(..)` that allows you to reset the position. I would reset it to the type root's source range and perform this logic there. It might be slightly longer because you'd copy over a new while loop. Then you could add logic to add the folding range.

Reply:
Ok, I've thought about it. Let's keep it as-is so you avoid passing in the values. I would just add a comment making it clear `ITerminalSymbols.TokenNameimport` only deals with the import range for `.class` files.